crates/nu-command/src/system/exec.rs

```diff
@@ -35,7 +32,66 @@ On Windows based systems, Nushell will wait for the command to finish and then e
         call: &Call,
         _input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
-        exec(engine_state, stack, call)
+        let cwd = engine_state.cwd(Some(stack))?;
+
+        // Find the absolute path to the executable. If the command is not
+        // found, display a helpful error message.
+        let name: Spanned<String> = call.req(engine_state, stack, 0)?;
+        let executable = {
+            let paths = nu_engine::env::path_str(engine_state, stack, call.head)?;
+            let Some(executable) = crate::which(&name.item, &paths, &cwd) else {
+                return Err(crate::command_not_found(
+                    &name.item,
+                    call.head,
+                    engine_state,
+                    stack,
+                ));
+            };
+            executable
+        };
+
+        // Create the command.
+        let mut command = std::process::Command::new(executable);
+
+        // Configure PWD.
+        command.current_dir(cwd);
+
+        // Configure environment variables.
+        let envs = env_to_strings(engine_state, stack)?;
+        command.env_clear();
+        command.envs(envs);
+
+        // Configure args.
+        let args = crate::eval_arguments_from_call(engine_state, stack, call)?;
+        command.args(args.into_iter().map(|s| s.item));
+
+        // Execute the child process, replacing/terminating the current process
+        // depending on platform.
+        #[cfg(unix)]
+        {
+            use std::os::unix::process::CommandExt;
+
+            let err = command.exec();
+            Err(ShellError::ExternalCommand {
+                label: "Failed to exec into new process".into(),
+                help: err.to_string(),
+                span: call.head,
+            })
+        }
+        #[cfg(windows)]
+        {
+            let mut child = command.spawn().map_err(|err| ShellError::ExternalCommand {
+                label: "Failed to exec into new process".into(),
+                help: err.to_string(),
+                span: call.head,
+            })?;
+            let status = child.wait().map_err(|err| ShellError::ExternalCommand {
+                label: "Failed to wait for child process".into(),
+                help: err.to_string(),
+                span: call.head,
+            })?;
+            std::process::exit(status.code().expect("status.code() succeeds on Windows"))
```

Comment:
Isn't this going to panic if we don't get an exit code? I'm not sure how well behaved every windows app is. Wondering if the original `.unwrap_or(0)` would be better here.

Reply:
It doesn't depend on the behavior of the child process. Here's how `status.code()` is implemented on Windows:

```
pub fn code(&self) -> Option<i32> {
    Some(self.0 as i32)
}
```

`status.code()` only fails if the child process is interrupt by a Unix signal. On Windows there's no signal, and it always succeeds.

Reply:
interesting, thanks for the clarification.